From a21924aa4b7429f0b85578976977efdc7124c639 Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 24 Sep 2014 15:21:36 -0400
Subject: [infiniband] ocrdma: Move ocrdma_inetaddr_event outside of "#if CONFIG_IPV6"

Message-id: <3f7eb996013acb29d9d4af122e1be7e205e041ea.1411571481.git.dledford@redhat.com>
Patchwork-id: 96609
O-Subject: [Patch RHEL7 01/48] RDMA/ocrdma: Move ocrdma_inetaddr_event outside of "#if CONFIG_IPV6"
Bugzilla: 1081067 1088577
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Roland Dreier <roland@purestorage.com>

This fixes the build if IPV6 isn't enabled.

Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 31ab8acbf6618c89fec77f7706df7daaa319feb5)
Signed-off-by: Doug Ledford <dledford@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/ocrdma/ocrdma_main.c b/drivers/infiniband/hw/ocrdma/ocrdma_main.c
index 47187bf..2ca86ca 100644
--- a/drivers/infiniband/hw/ocrdma/ocrdma_main.c
+++ b/drivers/infiniband/hw/ocrdma/ocrdma_main.c
@@ -174,6 +174,10 @@ static int ocrdma_inetaddr_event(struct notifier_block *notifier,
  return ocrdma_addr_event(event, netdev, &gid);
 }
 
+static struct notifier_block ocrdma_inetaddr_notifier = {
+ .notifier_call = ocrdma_inetaddr_event
+};
+
 #if IS_ENABLED(CONFIG_IPV6)
 
 static int ocrdma_inet6addr_event(struct notifier_block *notifier,
@@ -185,10 +189,6 @@ static int ocrdma_inet6addr_event(struct notifier_block *notifier,
  return ocrdma_addr_event(event, netdev, gid);
 }
 
-static struct notifier_block ocrdma_inetaddr_notifier = {
- .notifier_call = ocrdma_inetaddr_event
-};
-
 static struct notifier_block ocrdma_inet6addr_notifier = {
  .notifier_call = ocrdma_inet6addr_event
 };
-- 
1.7.1